### PR TITLE
Add hawk authentication for the airflow api

### PR DESCRIPTION
**Pull request by niross**
### Description of change

Protect the airflow api behind hawk authentication

### Checklist

* [x] Have tests been added to cover any changes?
* [x] Has the [CHANGELOG](https://github.com/uktrade/data-flow/blob/master/CHANGELOG.md) been updated?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: bin/paas-wrapper.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@ export PYTHONPATH=/app:${PYTHONPATH}
 export AIRFLOW_HOME=/home/vcap/app/airflow
 export DEBUG=False
 
-export AIRFLOW__API__AUTH_BACKEND=airflow.api.auth.backend.deny_all
+export AIRFLOW__API__AUTH_BACKEND=dataflow.api_auth_backend
```
Comment (michalc) [inline bin/paas-wrapper.sh:18]: It looks like there are tests of the `requires_authentication` function against mock views. Can there be a test of an actual API call against a real view? [For example, it looks like if the above line is changed back, all the tests would still pass?]

Comment (niross) [reply bin/paas-wrapper.sh:18]: I can look into it but I felt like as the views and auth are baked into airflow/flask (and covered in their tests) it wasn't really necessary. 

We don't currently run any integration tests on the airflow web ui.

File: dataflow/config.py [hunk 2]
```diff
@@ -103,3 +103,9 @@
 
 PEOPLE_FINDER_BASE_URL = os.environ.get('PEOPLE_FINDER_BASE_URL')
 PEOPLE_FINDER_PRIVATE_KEY = os.environ.get('PEOPLE_FINDER_PRIVATE_KEY')
+
+DATA_STORE_UPLOADER_SENDER_HAWK_CREDENTIALS = (
+    os.environ.get('DATA_STORE_UPLOADER_SENDER_HAWK_ID'),
+    os.environ.get('DATA_STORE_UPLOADER_SENDER_HAWK_KEY'),
+)
+AIRFLOW_API_HAWK_CREDENTIALS = (DATA_STORE_UPLOADER_SENDER_HAWK_CREDENTIALS,)
```
Comment (samuelhwilliams) [inline dataflow/config.py:111]: Any reason not to just use a straight-up dict here? Seems like all we do is lookup by sender ID?

Comment (niross) [reply dataflow/config.py:111]: No reason except I changed tack half way though, I'll change it :+1:

File: bin/paas-wrapper.sh (hunk 1 above)
Comment (niross) [reply bin/paas-wrapper.sh:18]: Integration tests added :+1:

File: dataflow/config.py [hunk 3]
```diff
@@ -103,3 +103,13 @@
 
 PEOPLE_FINDER_BASE_URL = os.environ.get('PEOPLE_FINDER_BASE_URL')
 PEOPLE_FINDER_PRIVATE_KEY = os.environ.get('PEOPLE_FINDER_PRIVATE_KEY')
+
+DATA_STORE_UPLOADER_SENDER_HAWK_CREDENTIALS = (
+    os.environ.get('DATA_STORE_UPLOADER_SENDER_HAWK_ID'),
+    os.environ.get('DATA_STORE_UPLOADER_SENDER_HAWK_KEY'),
+)
```
Comment (samuelhwilliams) [inline dataflow/config.py:110]: Can this go now?

File: tests/test_api_auth.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
+from unittest import mock
```
Comment (samuelhwilliams) [inline tests/test_api_auth.py:1]: Do we need to move this file into `tests/unit` as well?

File: dataflow/config.py (hunk 3 above)
Comment (niross) [reply dataflow/config.py]: Yep, done

File: tests/test_api_auth.py (hunk 4 above)
Comment (niross) [reply tests/test_api_auth.py]: good catch yes i've moved it :+1:

File: tests/integration/test_api.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+            credentials={'id': hawk_id, 'key': hawk_key, 'algorithm': 'sha256'},
+            url=url,
+            method='GET',
+            content='',
+            content_type='',
+        )
+        response = client.get(
+            url, headers={'Authorization': sender.request_header, 'Content-Type': ''}
+        )
+        assert response.status_code == 200
```
Comment (michalc) [inline tests/integration/test_api.py:46]: Can there be a test just like this one, but where the key is wrong and asserts there is a 401/403? [Ultimately that's my main concern... not that it works with the right key, but that it doesn't with the wrong key]

**Pull request merged**